src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -538,6 +539,7 @@ export class TerminalService extends Disposable implements ITerminalService {
 		this._register(this.onAnyInstanceProcessIdReady(() => this._saveState()));
 		this._register(this.onAnyInstanceTitleChange(instance => this._updateTitle(instance)));
 		this._register(this.onAnyInstanceIconChange(e => this._updateIcon(e.instance, e.userInitiated)));
+		// this._register(this.onAnyInstanceShellTypeChanged(e => this._updateShellType(e))); // Do I need something like this for onAnyInstanceShellTypeChanged?
```

Comment:
No I don't think so, the above is mainly for saving layout state so it can be restored after reload. If there's a problem with that and shell type we would fix in a different PR with a bug report.